bundles/product-list-search-rest-api/src/FondOfOryx/Zed/ProductListSearchRestApi/Persistence/ProductListSearchRestApiRepository.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace FondOfOryx\Zed\ProductListSearchRestApi\Persistence;
+
+use ArrayObject;
+use Generated\Shared\Transfer\ProductListCollectionTransfer;
+use Orm\Zed\ProductList\Persistence\Map\SpyProductListTableMap;
+use Orm\Zed\ProductList\Persistence\SpyProductListQuery;
+use Propel\Runtime\ActiveQuery\Criteria;
+use Propel\Runtime\ActiveQuery\ModelCriteria;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+
+/**
+ * @codeCoverageIgnore
+ *
+ * @method \FondOfOryx\Zed\ProductListSearchRestApi\Persistence\ProductListSearchRestApiPersistenceFactory getFactory()
+ */
+class ProductListSearchRestApiRepository extends AbstractRepository implements ProductListSearchRestApiRepositoryInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\ProductListCollectionTransfer $productListCollectionTransfer
+     *
+     * @return \Generated\Shared\Transfer\ProductListCollectionTransfer
+     */
+    public function searchProductList(
+        ProductListCollectionTransfer $productListCollectionTransfer
+    ): ProductListCollectionTransfer {
+        $productListQuery = $this->getBaseQuery($productListCollectionTransfer);
+
+        $productListQuery = $this->addFulltextSearchFields($productListQuery, $productListCollectionTransfer);
+        $productListQuery = $this->addSort($productListQuery, $productListCollectionTransfer);
+        $productListQuery = $this->preparePagination($productListQuery, $productListCollectionTransfer);
+
+        $productList = $this->getFactory()
+            ->createProductListMapper()
+            ->mapEntityCollectionToTransfers($productListQuery->find());
+
+        return $productListCollectionTransfer->setProductLists(new ArrayObject($productList));
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListCollectionTransfer $productListCollectionTransfer
+     *
+     * @return \Orm\Zed\ProductList\Persistence\SpyProductListQuery
+     */
+    protected function getBaseQuery(ProductListCollectionTransfer $productListCollectionTransfer): SpyProductListQuery
```

Comment:
Remove unused paramater...

Reply:
@daniel-rose Thanx for the comment. I've cleaned the paramters.